src/client/MealsWithID.js

```diff
@@ -0,0 +1,46 @@
+import React, { useEffect } from 'react'
+import { useParams } from "react-router-dom";
+import ShowReservations from './ShowReservations.js'
+
+import images from './images.js'
+import './MealsWithID.css'
+
+const API = "/api/meals"
+
+
+const MealsWithId = ({ meals, setMeals }) => {
+
+    useEffect(() => {
+        async function DataFetched() {
```

Comment:
Its ok try create a function to isolate the data fetching here (DataFetched). Think a bout the name though.
In that case I think you should await it and have it return the data and then call setMeals outside that function. You would also want to to use a try/catch to handle errors.